python/pyarrow/_compute.pyx

```diff
@@ -907,7 +1023,23 @@ cdef class _ReplaceSubstringOptions(FunctionOptions):
 
 
 class ReplaceSubstringOptions(_ReplaceSubstringOptions):
-    def __init__(self, pattern, replacement, *, max_replacements=-1):
+    """
+    Options for replacing matched substrings.
+
+    Parameters
+    ----------
+    pattern : str
+        Substring pattern to look for inside input values.
+    replacement : str
+        What to replace the pattern with.
+    max_replacements : int or None, default None
+        If given, the maximum number of strings to replace in each
+        input value.
```

Comment:
If not given, there is no limit in number of replacements?

Reply:
For SplitOptions, you used "(unlimited if None)"

Reply:
Right, I'll adapt to this one.